### PR TITLE
fix(DB): missing Lunar Festival GOs and creatures

### PR DESCRIPTION
**Pull request by LannyE**
Major Cities are missing Lunar Festival gameobjects and NPCs/Creatures
- Darnassus:  Added the entire Festival location  (gameobjects/creatures)
- Shattrath:  Added both Festival locations  (gameobjects/creatures)
- Dalaran:  Added the entire Festival location  (gameobjects/creatures)
- Silvermoon:  Added the entire Festival location  (gameobjects/creatures)
- Exodar: Added the entire Festival location (gameobjects/creatures)
- Exodar: Removed New Year's event objects from the Lunar Event (Festive 
Kegs, Party Tables, and extra Haybales)

Darnassus missing objects and NPCs mentioned as TODO in issue #4354

Pulled this data from the Trinitycore database.

Note: Shattrath gameobjects had two objects that had GUIDs already assigned
in the Azerothcore database.  I continued on with the guids from the issue #4354 
fix, using 241016 and 241017.

## Changes Proposed:
- Darnassus:  Add the entire Lunar Festival location  (gameobjects/creatures)
- Shattrath:  Add both Lunar Festival locations  (gameobjects/creatures)
- Dalaran:  Add the entire Lunar Festival location  (gameobjects/creatures)
- Silvermoon:  Add the entire Lunar Festival location  (gameobjects/creatures)
- Exodar: Add the entire Lunar Festival location (gameobjects/creatures)
- Exodar: Remove New Year's event objects from the Lunar Event (Festive 
Kegs, Party Tables, and extra Haybales)

## Issues Addressed:
- The Major Cities included in this fix are missing their entire Lunar Festival location.   Shattrath has two locations that are missing. 

## Tests Performed:
- Merged the SQL and visited each location to show the NPCs and gameobjects show up. 
- All locations show up. 

## How to Test the Changes:

- Import the SQL code.
- Visit each of the Cities mentioned in the changes proposed.
- Verify the Lunar Festival Event NPCs and their objects are present.

## Known Issues and TODO List:

- [Exodar's Commoners are duplicated while the Lunar Event is on.  Looking into a fix.  Booty bay has the same problem mentioned in #4330]
- [ ] 

## Target Branch(es):
- [x] Master

 
## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here in the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

### CODE REVIEW

**Comment by LannyE** [comment]
Here are screenshots of the location after the fix:
image
image
image
image
image

**Comment by Kitzunu** [comment]
I have no time to check myself atm, so pls check so the GUIDs are free :)

Gj👍

File: data/sql/updates/pending_db_world/rev_1611875417271883500.sql [hunk 1]
```diff
@@ -0,0 +1,685 @@
+INSERT INTO `version_db_world` (`sql_rev`) VALUES ('1611875417271883500');
+
+/* Darnassus Missing Objects and Creatures
+*/
+DELETE FROM `gameobject` WHERE `guid` IN(11997,82973,82974,82975,82976,82977,82978,82985,82994,82995,82996,82997,82998,82999,83000,83001,83002,83003,83004,83005,83006,83007,83008,83009,83010,83011,83012,83013,83014,83015,83016,83017,7183,7185,7238,7252,7254,7258);;
+DELETE FROM `game_event_gameobject` WHERE `eventEntry` = 7 and `guid` IN(11997,82973,82974,82975,82976,82977,82978,82985,82994,82995,82996,82997,82998,82999,83000,83001,83002,83003,83004,83005,83006,83007,83008,83009,83010,83011,83012,83013,83014,83015,83016,83017,7183,7185,7238,7252,7254,7258);;
+DELETE FROM `creature` WHERE `guid` IN(85543,85541,85539,85535,85536,85537);
+DELETE FROM `game_event_creature` WHERE `eventEntry` = 7 and `guid` IN(85543,85541,85539,85535,85536,85537);
```
Comment (FrancescoBorzi) [inline data/sql/updates/pending_db_world/rev_1611875417271883500.sql:8]: missing spaces after `IN`

File: data/sql/updates/pending_db_world/rev_1611875417271883500.sql [hunk 2]
```diff
@@ -0,0 +1,685 @@
...
+DELETE FROM `gameobject` WHERE `guid` IN(79786,79772,79767,79768,79769,79770,79771,79773,79774,79783,79784,79785,79788,79791,79792,79795,79796,79797,79798,79799,79800,79801,79802,79818,79819,79820,79821,79822,79827,79828,79829,79830,79834,79835,79839,79840,79844,79848,79850,79851,79852,79853,79854,79855,79856,79857,79858,79859,79871,79872,79873,79874,241016,241017,7199,7244,7247,7248,7250,7251,7261,7268,79787,79789,79790,79803,79804,79805,79806,79807,79808,79809,79823,79826,79831,79836,79841,79845,79846,79860,79861,79862,79863,7175,7241,7245);
+DELETE FROM `game_event_gameobject` WHERE `eventEntry` = 7 and `guid` IN(79786,79772,79767,79768,79769,79770,79771,79773,79774,79783,79784,79785,79788,79791,79792,79795,79796,79797,79798,79799,79800,79801,79802,79818,79819,79820,79821,79822,79827,79828,79829,79830,79834,79835,79839,79840,79844,79848,79850,79851,79852,79853,79854,79855,79856,79857,79858,79859,79871,79872,79873,79874,241016,241017,7199,7244,7247,7248,7250,7251,7261,7268,79787,79789,79790,79803,79804,79805,79806,79807,79808,79809,79823,79826,79831,79836,79841,79845,79846,79860,79861,79862,79863,7175,7241,7245);
```
Comment (FrancescoBorzi) [inline data/sql/updates/pending_db_world/rev_1611875417271883500.sql:112]: same

File: data/sql/updates/pending_db_world/rev_1611875417271883500.sql [hunk 3]
```diff
@@ -0,0 +1,685 @@
...
+(7,97681),
+(7,97683);
+
+
+/* Dalaran Missing Objects and Creatures
+*/
+DELETE FROM `game_event_creature` WHERE `eventEntry` = 7 and `guid` IN(61988,61989,61990,61992,61993,61991);
+DELETE FROM `creature` WHERE `guid` IN(61988,61989,61990,61992,61993,61991);
+DELETE FROM `gameobject` WHERE `guid` IN(36,79673,79664,79652,79606,79605,79604,79603,79602,79601,79600,79593,79592,79591,79590,79589,79588,79587,79586,79585,79584,79692,79706,79719,79720,79723,79737,79744,79745,79748,79753,79754,79755,79756,79757,79758,79759,79760,79761,79762,79763,79764,79765,79766,7177,7196,7235,7246,7257,7262,7266);
+DELETE FROM `game_event_gameobject` WHERE `eventEntry` = 7 and `guid` IN(36,79673,79664,79652,79606,79605,79604,79603,79602,79601,79600,79593,79592,79591,79590,79589,79588,79587,79586,79585,79584,79692,79706,79719,79720,79723,79737,79744,79745,79748,79753,79754,79755,79756,79757,79758,79759,79760,79761,79762,79763,79764,79765,79766,7177,7196,7235,7246,7257,7262,7266);
```
Comment (FrancescoBorzi) [inline data/sql/updates/pending_db_world/rev_1611875417271883500.sql:324]: same

File: data/sql/updates/pending_db_world/rev_1611875417271883500.sql [hunk 4]
```diff
@@ -0,0 +1,685 @@
...
+DELETE FROM `game_event_gameobject` WHERE `eventEntry` = 7 and `guid` IN(91326,91328,91481,91482,91487,91488,91489,91490,91491,91492,91494,92045,92047,92048,92051,92052);
+
+
+
+/*Silvermoon Missing Objects and Creatures
+*/
+DELETE FROM `game_event_gameobject` WHERE `eventEntry` = 7 and `guid` IN(38,79775,79776,79777,79778,79779,79780,79781,79782,79793,79794,79810,79811,79812,79813,79814,79815,79816,79817,79824,79825,79832,79833,79837,79838,79842,79843,79847,79849,79864,79865,79866,79867,79868,79869,79870,79875,79876);
+DELETE FROM `gameobject` WHERE `guid` IN(38,79775,79776,79777,79778,79779,79780,79781,79782,79793,79794,79810,79811,79812,79813,79814,79815,79816,79817,79824,79825,79832,79833,79837,79838,79842,79843,79847,79849,79864,79865,79866,79867,79868,79869,79870,79875,79876);
+DELETE FROM `creature` WHERE `guid` IN(97689,97690,97691,97685,97694,97682);
+DELETE FROM `game_event_creature` WHERE `eventEntry` = 7 and `guid` IN(97689,97690,97691,97685,97694,97682);
```
Comment (FrancescoBorzi) [inline data/sql/updates/pending_db_world/rev_1611875417271883500.sql:588]: same

File: data/sql/updates/pending_db_world/rev_1611875417271883500.sql [hunk 5]
```diff
@@ -0,0 +1,685 @@
...
+(7,79849),
+(7,79864),
+(7,79865),
+(7,79866),
+(7,79867),
+(7,79868),
+(7,79869),
+(7,79870),
+(7,79875),
+(7,79876);
```
Comment (FrancescoBorzi) [inline data/sql/updates/pending_db_world/rev_1611875417271883500.sql:685]: end line a the end of file

**Comment by FrancescoBorzi** [comment]
@LannyE thanks for your contribution, have you checked that the GUIDs from your PR are not already used in AC ?

**Comment by LannyE** [comment]
I continued any new GUIDS (GUIDS made to resolve conflicts between Trintycore GUIDS and Azerothcore GUIDS) from the last one used on fix #4354 which was made on Tuesday the 25th.  I think that continuing from that fix is the correct way because the two new gameobjects IDs are the same objects added in that fix.  These two objects are for the two Shattrath Greater Moonlight Spell Focus circles.  That would make for consistency.  I merged this with my server running the Azerothcore branch from Monday morning (AzerothCore rev. 3d60ce8e6e3f+ 2021-01-24 21:21:48 -0500 (master branch)).  It merged fine.  I went through the latest world DB updates from the most recent branch and didn't see any conflicts.   All other GUIDS being added were clear and came from the Trinitycore database and pieced in nicely.

**Comment by FrancescoBorzi** [comment]
> I continued any new GUIDS (GUIDS made to resolve conflicts between Trintycore GUIDS and Azerothcore GUIDS) from the last one used on fix #4354 which was made on Tuesday the 26th. I think that continuing from that fix is the correct way because the two new gameobjects IDs are the same objects added in that fix. These two objects are for the two Shattrath Greater Moonlight Spell Focus circles. That would make for consistency. I merged this with my server running the Azerothcore branch from Monday morning (AzerothCore rev. [3d60ce8](https://github.com/azerothcore/azerothcore-wotlk/commit/3d60ce8e6e3fbc110ca1d02f1e5074dfbfc1b018)+ 2021-01-24 21:21:48 -0500 (master branch)). It merged fine. I went through the latest world DB updates from the most recent branch and didn't see any conflicts. All other GUIDS being added were clear and came from the Trinitycore database and pieced in nicely.

@LannyE are you sure you've checked all these GUIDs ? This gives me a result:

```sql
SELECT * FROM `game_event_gameobject` WHERE `eventEntry` = 7 and `guid` IN (11997);
```

is it intended to replace it?

**Comment by LannyE** [comment]
GUID 11997 has no Gameobject in Azerothcore's database.    Trintycore has the Greater Moonlight Spell Focus ID 180876 for the six classic Major Cities in gameobject GUIDs 11995 - 12000.  For some reason Azerothcore has them moved to GUIDs 241000, 241002 -241006.    So in Azerothcore gameobject GUIDs 11995 - 12000 do not exist.  I religiously backup my databases and when I began with Azerothcore on Dec 22th GUIDs 11995 - 12000 do not exist either.  This tells me someone attempted to fix the Lunar Event way before I came.  

This is the mapping of the Greater Moonlight Spell Focus gameobject GUID migration from Trinitycore.  
11995->241004
11996 ->241005
11997->241002
11998->241003
11999->241000
12000->241006

???->241001  <-This may be a problem for #4354 as this is for one of the Moonglade Moonlight Major City return portals.  I assume a horde portal.  I will make a note in that issue.

Anyway that was a long answer to a short question.  GUID 11997 should be deleted from the game_event_gameobject table because it doesn't exist.  It does look like I made a mistake and am deleting it from gameobject also which does nothing cause it doesn't exist.  I will fix that.      

It also looks like I need to also remove 11995,11996,11998,11999,and 12000 from the game_event_gameobject table because they also do not exist.  This will clean up the game_event_gameobject table more of broken links.  
 
```SQL
DELETE FROM `game_event_gameobject` WHERE `eventEntry` = 7 and `guid` IN (11995,11996,11997,11998,11999,12000); 
```

**Comment by LannyE** [comment]
I made a new commit.  I appreciate your patience with this.  Let me know if I need to format this differently, or remove any sections.

File: data/sql/updates/pending_db_world/rev_1611875417271883500.sql [hunk 6]
```diff
@@ -1,9 +1,12 @@
 INSERT INTO `version_db_world` (`sql_rev`) VALUES ('1611875417271883500');
 
+/*Cleanup GUIDs that do not exist - These GUIDs were migrated to GUID 241000 and 241002 -241006*/
+DELETE FROM `game_event_gameobject` WHERE `eventEntry` = 7 and `guid` IN (11995,11996,11997,11998,11999,12000);
```
Comment (FrancescoBorzi) [inline data/sql/updates/pending_db_world/rev_1611875417271883500.sql:4]: but they exist for me. Is it intended to delete them in this PR?

image

Comment (Kitzunu) [reply data/sql/updates/pending_db_world/rev_1611875417271883500.sql:4]: No shin. Check gameobject instead. This table is not relevant if the GUID does not exist in gameobject

Comment (FrancescoBorzi) [reply data/sql/updates/pending_db_world/rev_1611875417271883500.sql:4]: ah right

**Review by FrancescoBorzi — APPROVED** [review]

**Comment by FrancescoBorzi** [comment]
looks good to me, thanks @LannyE for your contribution!

PS: what's your username on Discord?

**Comment by LannyE** [comment]
Thank you.  My discord name is lanny.

File: data/sql/updates/pending_db_world/rev_1611875417271883500.sql [hunk 7]
```diff
@@ -0,0 +1,688 @@
+INSERT INTO `version_db_world` (`sql_rev`) VALUES ('1611875417271883500');
+
+/*Cleanup GUIDs that do not exist - These GUIDs were migrated to GUID 241000 and 241002 -241006*/
+DELETE FROM `game_event_gameobject` WHERE `eventEntry` = 7 and `guid` IN (11995,11996,11997,11998,11999,12000); 
+
+/* Darnassus Missing Objects and Creatures
+*/
+DELETE FROM `gameobject` WHERE `guid` IN (82973,82974,82975,82976,82977,82978,82985,82994,82995,82996,82997,82998,82999,83000,83001,83002,83003,83004,83005,83006,83007,83008,83009,83010,83011,83012,83013,83014,83015,83016,83017,7183,7185,7238,7252,7254,7258);;
+DELETE FROM `game_event_gameobject` WHERE `eventEntry` = 7 and `guid` IN (82973,82974,82975,82976,82977,82978,82985,82994,82995,82996,82997,82998,82999,83000,83001,83002,83003,83004,83005,83006,83007,83008,83009,83010,83011,83012,83013,83014,83015,83016,83017,7183,7185,7238,7252,7254,7258);;
```
Comment (pklloveyou) [inline data/sql/updates/pending_db_world/rev_1611875417271883500.sql:9]: Delete one ```;```

**Comment by LannyE** [comment]
Changes made.  Thank you.

**Comment by pklloveyou** [comment]
Tested

**Comment by FrancescoBorzi** [comment]
thanks @LannyE for your contribution and all the people involved in reviewing/testing!

**Pull request merged**